src/providers/realtime-provider.js

```diff
@@ -99,166 +88,119 @@ export default class RealtimeProvider {
         };
     }
 
-    sendSubscribeMessage(subscriptionDetails) {
-        let domainObject = subscriptionDetails.domainObject;
-        let message = MESSAGES.SUBSCRIBE[domainObject.type](subscriptionDetails);
-
-        this.sendOrQueueMessage(message);
-    }
-
-    sendUnsubscribeMessage(subscriptionDetails) {
-        let message = MESSAGES.UNSUBSCRIBE(subscriptionDetails);
-
-        this.sendOrQueueMessage(message);
+    _connect() {
+        this.socket.onopen(this._onopen.bind(this));
+        this.socket.onmessage(this._onmessage.bind(this));
     }
 
-    reconnect() {
-        if (this.reconnectTimeout) {
-            return;
-        }
-
-        this.reconnectTimeout = setTimeout(() => {
-            this.connect();
-            delete this.reconnectTimeout;
-        }, FALLBACK_AND_WAIT_MS[this.currentWaitIndex]);
-
-        if (this.currentWaitIndex < FALLBACK_AND_WAIT_MS.length - 1) {
-            this.currentWaitIndex++;
-        }
-    }
-
-    sendOrQueueMessage(request) {
-        if (this.connected) {
+    _flushQueue() {
```

Comment:
The queuing logic needs to be moved into the yamcs-web-socket, because the realtime fault provider is going to need this too. This is needed when subscription requests arrive when the websocket is not connected (either because it's still handshaking, or it has dropped out - which happens regularly).